src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,171 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int ONE = 1;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = createNewNode(value);
+        if (head == null) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+        }
+        tail = newNode;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        if (index == size()) {
+            add(value);
+        } else {
+            Node<T> newNode = createNewNode(value);
+            if (index == 0) {
+                addFirst(newNode);
+            } else {
+                Node<T> current = getNodeAtIndex(index - ONE);
+                insertAfter(current, newNode);
+            }
+        }
```

Comment:
if you slightly reorder this method you won't need separate method for index check and will have less tabulations for majority of code
```suggestion
        if (index == size()) {
            add(value);
            return;
        }
        checkIndex(index);
        Node<T> newNode = createNewNode(value);
        if (index == 0) {
            addFirst(newNode);
        } else {
            Node<T> current = getNodeAtIndex(index - ONE);
            insertAfter(current, newNode);
        }
```